[docs] Added copy button Docs markdown's code blocks

huseyinbuyukdere opened the pull request:
Preview: https://deploy-preview-27308--material-ui.netlify.app/components/buttons

Added copy button to code blocks in markdown. I redirect markdown code to highlightedCode component and edited highlightedCode component to show a copy button. 

Seperated some sh code blocks to copy easly like yarn and npm install commands.  It is working like below 

![copy](https://user-images.githubusercontent.com/39594189/125846777-53f38e4e-7670-4e25-b932-deec936a2ab6.gif)

eps1lon reviewed (changes requested):
We already have a button to copy the source of the demos.

For inline blocks, it'd be nice but then only add the button to these blocks. It should also be permanently visible, not on hover.

huseyinbuyukdere commented:
Hi @eps1lon , thanks for review, I did what you offer, but the inline blocks copy button seems not very good with the demo source code copy button like below.

![image](https://user-images.githubusercontent.com/39594189/126060184-2b9a6d83-a3c5-4bc0-babb-1a665496ebf4.png)

If I changed demo copy button  like below , it seems better. I removed copy button to code blocks in demos, added like inline blocks. 

![image](https://user-images.githubusercontent.com/39594189/126060369-57439ebc-60dc-4061-b8bb-75e494f354bd.png)

By the way, I want to ask you how should I place the copy button when the code blocks fill all width like below. If I placed copy button, it is overlapping with the code. How should I place that.

![image](https://user-images.githubusercontent.com/39594189/126060292-520f3fe8-66fc-4d7b-abb9-225c74f09c68.png)

eps1lon commented:
> Hi @eps1lon , thanks for review, I did what you offer, but the inline blocks copy button seems not very good with the demo source code copy button like below.

This seems fine to me. I find splitting actions into toolbar and copy-code worse.

huseyinbuyukdere commented:
I  put it copy button to toolbox slowly, I removed from demo code blocks, It is working for only inline code blocks.

huseyinbuyukdere commented:
@eps1lon  Do I need to do something ?

eps1lon commented on docs/src/pages/components/about-the-lab/about-the-lab-de.md:
````diff
@@ -20,20 +20,18 @@ Damit eine Komponente zum Kern (Core) übergehen kann, werden folgende Kriterien
 npm install @material-ui/lab
-
-// mit yarn
+```
````
The changes to translation files are unnecessary since they'd be reverted immediately. Only change the english markdown files if required.

huseyinbuyukdere replied:
Ok, I will change only english documents, but there will be a problem while copying the install blocks from other languages doc. Because yarn and npm install commands will be copied together.

![image](https://user-images.githubusercontent.com/39594189/128556168-3d468e1a-efee-4952-a29c-32a6a20c4225.png)

eps1lon commented on docs/src/modules/components/HighlightedCode.js:
```diff
@@ -1,29 +1,72 @@
 
 const HighlightedCode = React.forwardRef(function HighlightedCode(props, ref) {
-  const { code, language, ...other } = props;
+  const { code, language, isCopyButtonEnabled, ...other } = props;
```
I suspect we want to enable it by default since most usages are for inline code. Only the demo would disable it, right?

huseyinbuyukdere replied:
Expect demos, all inline code blocks will have copy button. On the other hand some pages using Highlighted code components will have copy button too, because I changed that manually, not demo component which we talked about. 

I suspected about that too, but when I saw the github improvement about these blocks - they did for all github markdown view - I though it could not be problem. 

Only migration blocks may be problem, you know migration inline blocks only show that the differences but we are adding copy button too. It seems a little strange. But github has same strange behaviour too like below. They are adding copy button all inline code blocks directly. 

```
 warning = {
-  main: orange[500],
+  main: "#ED6C02", // orange[400] in "dark" mode

-  light: orange[300],
+  light: orange[500], // orange[300] in "dark" mode

-  dark: orange[700],
+  dark: orange[900], // orange[700] in "dark" mode
 }
```

eps1lon commented:
Really appreciate the work. Though the double pass render and encoding really hurts maintainability for a very critical part of our documentation. We may revisit this in the future when we overhaul how we generate documentation. 

For now people can already copy actual demos. For inline code there's rarely a need.
